src/dnf-plugins/product-id/test-product-id.c

```diff
@@ -93,11 +94,18 @@ typedef struct {
 } handleFixture;
 
 void setup(handleFixture *fixture, gconstpointer testData) {
+    GError *error = NULL;
     (void)testData;
     fixture->dnfContext = dnf_context_new();
+    dnf_context_setup(fixture->dnfContext, NULL, &error);
     PluginMode mode = PLUGIN_MODE_CONTEXT;
-    // This is never explicitly called
-    fixture->handle = pluginInitHandle(1, mode, (void*)fixture->dnfContext);
+    // Following method is called in plugin, but it should never been explicitly called
+    // fixture->handle = pluginInitHandle(1, mode, (void*)fixture->dnfContext);
+    fixture->handle = malloc(sizeof(PluginHandle));
```

Comment:
Considering this code uses Glib, I'd say to use `g_malloc()` directly as it calls `abort()` on allocation failures. Other Glib bits in this code will do this internally anyway, so it will not make things worse.

Reply:
Yeah, good point. It is only unit test, but why not.